From 6d288ad39ceec639779b58d5a585ead599fe2997 Mon Sep 17 00:00:00 2001
From: Richard Purdie <richard.purdie@linuxfoundation.org>
Date: Fri, 11 Sep 2026 09:04:05 +0100
Subject: [PATCH] An easy to reproduce make race:

$ make libswscale/x86/uops_macros.gen.asm
HOSTCC	libswscale/x86/uops_macros.gen.asm
cc1: fatal error: opening output file libswscale/x86/uops_macros.gen.asm: No such file or directory

It needs the missing directory creation dependency adding.

Upstream-Status: Submitted [Sent to https://lists.ffmpeg.org/archives/list/ffmpeg-devel@ffmpeg.org/ 2026/9/11]
Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>
---
 libswscale/x86/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libswscale/x86/Makefile b/libswscale/x86/Makefile
index 0f7a3db..f56a5a4 100644
--- a/libswscale/x86/Makefile
+++ b/libswscale/x86/Makefile
@@ -1,3 +1,5 @@
+OUTDIRS += $(SUBDIR)x86
+
 OBJS                            += x86/rgb2rgb.o                        \
                                    x86/swscale.o                        \
                                    x86/yuv2rgb.o                        \
@@ -31,6 +33,6 @@ $(SUBDIR)x86/ops_common.o: $(SUBDIR)x86/uops_macros.gen.asm
 $(SUBDIR)x86/ops_int.o: $(SUBDIR)x86/uops_macros.gen.asm
 $(SUBDIR)x86/ops_float.o: $(SUBDIR)x86/uops_macros.gen.asm
 $(SUBDIR)x86/uops_macros.gen.asm: $(SRC_PATH)/libswscale/x86/uops_macros.asm.h \
-                                  $(SRC_PATH)/libswscale/uops_macros.h
+                                  $(SRC_PATH)/libswscale/uops_macros.h | $(SUBDIR)x86
 	$(HOSTCC) $(HOSTCC_E) $(CPPFLAGS) $<
 endif
